pydantic_ai_slim/pydantic_ai/toolsets/_dynamic.py

```diff
@@ -17,39 +16,92 @@
 """A sync/async function which takes a run context and returns a toolset."""
 
 
-@dataclass
+class ToolsetRunStep(TypedDict, Generic[AgentDepsT]):
```

Comment:
- Could this be a `dataclass`?
- Can we come up with a more clear name? Something with "state" in it